vyper/builtin_functions/functions.py

```diff
@@ -2621,6 +2689,7 @@ def evaluate(self, node):
     "pow_mod256": PowMod256(),
     "uint2str": Uint2Str(),
     "sqrt": Sqrt(),
+    "sqrt_solmate": SqrtSolmate(),
```

Comment:
why not update the original Sqrt instead?

Reply:
That was the original idea indeed, but did not want to introduce that breaking change without prompt from a core contributor.

I'll make the changes to the original sqrt then.

Reply:
i don't think we should get rid of the original sqrt. sqrt on decimals is pretty useful, especially once we open up the decimals type to be generic.

Reply:
Yeah, but sounds like this should basically expand the types that sqrt works with, so if it's decimals it uses the previous and if it's integers it uses this one

Reply:
i think if it's possible, use the same algorithm for both types. there will be fewer gas surprises that way

Reply:
@charles-cooper It doesn't look possible since exponentiation is not allowed for decimal type. I do like the idea @fubuloubu mentioned, where `sqrt(x: decimal)` would just yield what it does right now and `sqrt(x: uint256)` would use the optimised version. 

Regarding gas surprises: is that really an issue? I don't expect users to use the decimal type at all.

Reply:
discussed offline -- the gas difference is ok. @bout3fiddy added a note that we will try to reify the code paths later